src/main/java/ar/com/synergian/wagongit/Utils.java

```diff
@@ -11,15 +11,14 @@
 	private Utils() {
 	}
 
-	public static File createCheckoutDirectory(String path, boolean isPermanent) throws GitException {
+	public static File createCheckoutDirectory(String path) throws GitException {
 		File dir;
-		if (isPermanent) {
-			dir = new File(path);
-		} else {
+		if (path.endsWith(".git")) {
 			dir = new File(System.getProperty("java.io.tmpdir"), "wagon-git-" + hashPath(path));
+		} else {
+			dir = new File(path);
```

Comment:
I'm not totally happy with this change. Let's forget path for a while, path is just a text transformation of the repo URI. What if the new argument just replaces the new File(java.io.tempdir) if present (not null)?